pandas/lib.pyx

```diff
@@ -1055,6 +1055,25 @@ def string_array_replace_from_nan_rep(ndarray[object, ndim=1] arr, object nan_re
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
+def object_array_decode_bytes(ndarray[object, ndim=1] arr, object encoding):
+    """Decode any instances of bytes to str in arr using the given encoding."""
+    if bytes == str: # in Python 2 these are the same and nothing needs to be done
+        return
+
+    cdef int length = arr.shape[0], i = 0
+    for i from 0 <= i < length:
+        if isinstance(arr[i], bytes):
+            arr[i] = arr[i].decode(encoding)
```

Comment:
again instead of writing this, simply use ``pandas.core.strings.str_decode`` it *already* takes an array, handles nulls and does the right thing. 

We have a tremendous amount of code already in pandas. So we really really try to avoid duplication.